pubsub/topics/publish_settings.go

```diff
@@ -39,14 +39,19 @@ func publishWithSettings(w io.Writer, projectID, topicID, msg string) error {
 	t.PublishSettings.CountThreshold = 10
 	t.PublishSettings.DelayThreshold = 100 * time.Millisecond
 
-	result := t.Publish(ctx, &pubsub.Message{Data: []byte(msg)})
-	// Block until the result is returned and a server-generated
-	// ID is returned for the published message.
-	id, err := result.Get(ctx)
-	if err != nil {
-		return fmt.Errorf("Get: %v", err)
+	for i := 0; i < 10; i++ {
+		result := t.Publish(ctx, &pubsub.Message{
+			Data: []byte("Message " + strconv.Itoa(i)),
+		})
+		// Block until the result is returned and a server-generated
+		// ID is returned for the published message.
+		id, err := result.Get(ctx)
```

Comment:
This currently blocks after every single Publish. Instead, we should probably use the same method as https://github.com/GoogleCloudPlatform/golang-samples/blob/master/pubsub/topics/publish_scale.go

Reply:
Does resolving the publish futures this way prevent batching? 
If so, the Python sample needs a rewrite as well.

Reply:
I believe so. Since we're guaranteed to get `id` populated after `Get`, the message has to have been published already.
I'm not sure about the Python library.

Reply:
Done

Reply:
Thank you! I will update the Python sample to use `.add_done_callback` today.